### ISSUE TITLE
[JAVA-2763] Investigate dropping Java 8 in favor of Java 11

### ISSUE THREAD

**Issue by adutra** [datastax/cassandra-quarkus#32]
<p>Imported from [JAVA-2763](https://datastax-oss.atlassian.net/browse/JAVA-2763), originally reported by @adutra on Wed, 6 May 2020 14:16:30 GMT.</p>
<hr />
<p>The Quarkus team [announced recently](https://quarkus.io/blog/quarkus-1-4-final-released/#java-8-deprecated) that they are going to deprecate Java 8 and drop support for it entirely starting with Quarkus 1.6, due in two months.</p> 
<p>Investigate if we shouldn't drop support for Java 8 too in the cassandra-quarkus-extension and switch to Java 11. Alternatively, we could keep supporting Java 8 while building and testing against Java 11.</p>

┆Issue is synchronized with this [Jira Task](https://datastax-oss.atlassian.net/browse/QUAR-14) by [Unito](https://www.unito.io)

**Comment by adutra** [comment]
Update: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/quarkus-dev/yzEjmYCFbwY/oW64kts3AQAJ.

It might be a bit premature to drop Java 8, so I'm removing this from 1.0.0-alpha1 milestone.

**Comment by adutra** [comment]
Noticed that the builds now spit this warning:

```
[WARNING] [io.quarkus.deployment.QuarkusAugmentor] Using Java versions older than 11 to build Quarkus applications is deprecated and will be disallowed in a future release!
```

**Comment by absurdfarce** [comment]
FWIW that warning has been there for some time @adutra .  Given that the Quarkus team has shifted to more of a wait-and-see approach on this topic I don't think there's much we can do about this.

**Comment by adutra** [comment]
I agree, but our extension is new in the market. Only projects starting now or later will ever use it. So the question is: does it make sense to start a greenfield project in 2020 based on Java 8?

**Comment by absurdfarce** [comment]
I don't know that I have a strong opinion either way, but if we moved to Java11 before Quarkus did we'd potentially cut _some_ users out from being able to use the extension.  Upgrading Java versions is always a big deal in larger deployments.  In a certain way this ties back to the earlier question about whether customers who were asking for Quarkus expect to keep up with the fairly rapid pace of Quarkus releases or not.  If customers are going to continue using, say, Quarkus 1.4.2 (picking a version at random) then forcing Java11 prevents those customers from using this extension.

Again, I don't have a strong opinion so I'm not arguing one way or the other.

**Comment by adutra** [comment]
See also this thread on the ML:

https://groups.google.com/g/quarkus-dev/c/7SZAM2BMb9c/m/KyVjbAWrAgAJ

And this wiki page:

https://github.com/quarkusio/quarkus/wiki/Why-Dropping-Java-8

**Comment by absurdfarce** [comment]
Note: the conversation (and wiki page) referenced by @adutra above were triggered by a [thread](https://camel.465427.n5.nabble.com/Re-HEADS-UP-Java-8-to-be-dropped-td5873176.html#none) on the Camel ML talking about their plans to move to Java11.  That thread doesn't provide a clear answer on "why are you requiring Java11" (which was the question posed by the Quarkus team on learning the news) but it did remind me of an interesting facet of this conversation to consider.  At some point in that thread the point was made that the alternate (better?) way to think about this was to ask "how long are we supporting Java8"?  Oracle is done providing free public updates to Java8.  Java11 is the next LTS.  It does make sense for people to move to that.

In short, I think I'm talking myself into what I take to be your position @adutra

**Comment by adutra** [comment]
Closing since Quarkus 2.x requires Java 11+ anyways.
